### ISSUE TITLE
Google Batch: Not possible to point to external source code

### ISSUE THREAD

**Issue by VasLem** [nextflow-io/nextflow#4956]
## Bug report 
Bringing up an issue, as noted [here](https://github.com/nextflow-io/training/discussions/328#discussion-6540560)
Long story short, Nextflow has the ability to support a `bin` folder, that can technically be seen in env var PATH during processes runtime. However, this is not the case when running on GCP, as the env var PATH variable is getting reset.  This leads to the files not being visible to the process. In particular, the following workflow:

```
process hello{
"""
echo $PATH
hello.sh
"""

}
workflow{
    hello()
}
```

produces an error after printing the $PATH:

```
/usr/local/bin:/usr/bin:/bin:/usr/games
```

and the error that the file hello.sh, with contents:

```
#!/usr/bin/env bash

echo "HELLO WORLD"
```

and with the +x flag, located in `bin` folder in the same directory as the workflow, cannot be found. [Here](https://github.com/nextflow-io/training/discussions/328#discussioncomment-9192525) you can find also the files contained in the workdir folder.

A workaround is to change the order of the injected commands, used to copy the files and update the PATH variable, within command.run as mentioned [here](https://github.com/nextflow-io/training/discussions/328#discussioncomment-9194853) . 

The occurring issue does not fall into the limitations ensued by not using [Wave](https://www.nextflow.io/docs/latest/module.html#module-binaries) as far as I can tell, the feature that has existed prior to version 22 (as it can be seen [here](https://training.nextflow.io/advanced/structure/) and [here](https://www.nextflow.io/docs/latest/sharing.html#bundling-executables-in-the-workflow)) . I have tested the `templates` functionality and it works as expected, however, although I could use this as an alternative, it prohibits any inter-dependencies between files that exist in the `bin` folder.

There are several issues for not allowing such a functionality, with the main ones being the impeded development of complex source codes, as there is currently no IDE supporting embedded Nextflow code AFAIK, and the inability to share code among different processes. Copying code around is error-prone and not something to be promoted by a modern tool as a good software approach.

**Comment by pditommaso** [comment]
Sorry but it's not clear what the bug is and how to replicate

**Comment by VasLem** [comment]
Hi Paolo, I updated the description with more verbosity, sorry for not following the template, if it is still not clear, please let me know.

**Comment by VasLem** [comment]
Hi Paolo, I don't think it is invalid, indeed I am referring to the first case. Can you not reproduce it in GCP?

**Comment by VasLem** [comment]
Thanks for the feedback, indeed I managed to work around the error. For posterity, I was printing $PATH and not \$PATH, so I was not receiving the PATH during the process runtime. I was able to call hello.sh, not by `./hello.sh` but by just `hello.sh`. Also, **in my actual workflow, which belongs in a folder structure, I had to symlink the bin directory, where the binaries were located, to that worflow's folder**, as anywhere else, .i.e., in the project root or even in the folder of the process using the binary, the binaries were not detected. Sym-linking folders is not a very well-organized way, particularly when you have multiple workflows using the same process that is dependent on a single binary, but it works.
